### PR TITLE
Add offset to TimeSeries

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
## Summary of changes

Sister PR to https://github.com/NeurodataWithoutBorders/pynwb/pull/1424, this suggests adding an `offset` attribute to all `TimeSeries` objects for additional support in converting data storage into proper units.

To convert the TimeSeries data field to the specified units, one would perform the operation

```python
data_in_units = data_in_storage * conversion + offset
```

## PR checklist for schema changes

- [x] Update the version string in `docs/format/source/conf.py`, `core/nwb.namespace.yaml`, and `/core/nwb.file.yaml`
  to the next version with the suffix "-alpha"
- [x] Add a new section in the release notes for the new version with the date "Upcoming"
- [x] Add release notes for the PR to `docs/format/source/format_release_notes.rst`

### CODE REVIEW

File: core/nwb.base.yaml [hunk 1]
```diff
@@ -124,6 +124,15 @@ groups:
...
       required: false
+   - name: offset
+      dtype: float32
+      default_value: 0.0
+      doc: Scalar to add to the data after scaling by 'conversion' to finalize its coercion
+        to the specified 'unit'. Two common examples of this include (a) data stored in an
+        unsigned type that requires a shift after scaling to re-center the data,
+        and (b) specialized recording devices that naturally cause a scalar offset with
+        respect to the true units.
+      required: false
```
Comment (rly) [inline core/nwb.base.yaml:135]: Please add a note to new line 147 with "offset"

File: core/nwb.icephys.yaml [hunk 2]
```diff
@@ -193,7 +193,7 @@ groups:
       value: volts
       doc: Base unit of measurement for working with the data. which is fixed to 'volts'.
         Actual stored values are not necessarily stored in these units. To access the data in these units,
-        multiply 'data' by 'conversion'.
+        multiply 'data' by 'conversion' and add 'offset'..
```
Comment (rly) [inline core/nwb.icephys.yaml]:
```suggestion
        multiply 'data' by 'conversion' and add 'offset'.
```

File: core/nwb.icephys.yaml [hunk 3]
```diff
@@ -116,7 +116,7 @@ groups:
       value: amperes
       doc: Base unit of measurement for working with the data. which is fixed to 'amperes'.
         Actual stored values are not necessarily stored in these units. To access the data in these units,
-        multiply 'data' by 'conversion'.
+        multiply 'data' by 'conversion' and add 'offset'..
```
Comment (rly) [inline core/nwb.icephys.yaml]:
```suggestion
        multiply 'data' by 'conversion' and add 'offset'.
```

File: core/nwb.icephys.yaml [hunk 4]
```diff
@@ -100,7 +100,7 @@ groups:
...
```
Comment (rly) [inline core/nwb.icephys.yaml]:
```suggestion
        multiply 'data' by 'conversion' and add 'offset'.
```

**Review by rly — CHANGES_REQUESTED** [review]

File: core/nwb.base.yaml (hunk 1 above)
Comment (CodyCBakerPhD) [reply core/nwb.base.yaml:135]: Done, and added to all other objects I can find.

**Review by rly — DISMISSED** [review]

**Review by rly — APPROVED** [review]

**Pull request merged**